src/pkg/util/nvidiautils/nvidiautils.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package nvidiautils
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/singularityware/singularity/src/pkg/sylog"
+)
+
+func nvidiaContainerCli() ([]string, []string, error) {
+	var strArray []string
+	var bindArray []string
+	var soID = regexp.MustCompile(".so")
+
+	// use nvidia-container-cli (if present)
+	command, err := exec.LookPath("nvidia-container-cli")
+	if err != nil {
+		return nil, nil, nil
+	}
+	cmd := exec.Command(command, "list", "--binaries", "--ipcs", "--libraries")
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, nil, nil
```

Comment:
If we're unable to execute the `nvidia-container-cli` command, we *should* return an error:

```
if err != nil {
    return nil, nil, fmt.Errof("Unable to execute ldconfig: %v", err)
}
```

We should only be ignoring errors in the case where `nvidia-container-cli` doesn't exist in path